apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -108,12 +119,81 @@ type WebstudioComponentDevProps = {
   components: Components;
 };
 
-export const WebstudioComponentDev = ({
-  instance,
-  instanceSelector,
-  children,
-  components,
-}: WebstudioComponentDevProps) => {
+/**
+ * Radix's VisuallyHiddenPrimitive.Root https://github.com/radix-ui/primitives/blob/main/packages/react/visually-hidden/src/VisuallyHidden.tsx
+ * component makes content from hidden elements accessible to screen readers.
+ * Using the same Content children, however, leads to duplicated React elements, breaking our 'isSelected' logic.
+ * To prevent this, we check if an instance is a descendant of VisuallyHiddenPrimitive.Root, and if so,
+ * we avoid rendering it.
+ */
+const useIsScreenReaderDescendant = (ref: RefObject<HTMLElement>) => {
+  const [isScreenReaderDescendant, setIsScreenReaderDescendant] =
+    useState(false);
+
+  useEffect(() => {
+    if (ref.current !== null) {
+      for (
+        let element: HTMLElement | null = ref.current;
+        element !== null;
+        element = element.parentElement
+      ) {
+        if (
+          element.style.overflow === "hidden" &&
+          element.style.clip === "rect(0px, 0px, 0px, 0px)" &&
+          element.style.position === "absolute" &&
+          element.style.width === "1px" &&
+          element.style.height === "1px"
+        ) {
+          setIsScreenReaderDescendant(true);
+          return;
+        }
+      }
+    }
+  }, [ref]);
+
+  return isScreenReaderDescendant;
+};
+
+/**
+ * For some components that are wrapped with Radix Slot components (where asChild=true),
+ * events are passed implicitly. We aim to merge these implicit events with the explicitly defined ones.
+ **/
+type ImplicitEvents = {
```

Comment:
is there any way to not have a logic that is specific to radix here? I don't know how, but maybe there is some way?

Reply:
Added comment about slots in other framework, they all behave the same, so it's not radix specific but how every Slot like wrapper works like

Reply:
but why do we have here only Click and Submit?

Reply:
why these 2 and nothing else?

Reply:
we overrided theese 2 in the code below to support links and forms.
Here are both just for simplicity of reading `composeEventHandlers` below, so coder can see that
we are merging 
implicit `onClick` from radix like component, with WebstudioComponentDev internal onClick which call userProps.onClick
But here can be any handler, as with our ts settings I was unable to define Partial on a record type. 
And comment below says about that `We ignore the remaining events because we currently detect events using the 'on' prefix.` and our inability to use Partial here